cpp/oneapi/dal/algo/kmeans/test/batch.cpp

```diff
@@ -0,0 +1,499 @@
+            for (std::int64_t i = 0; i < left_rows.get_count(); i++) {
+                const Float l = left_rows[i];
+                const Float r = right_rows[i];
+                if (l != r) {
+                    CAPTURE(l, r);
+                    FAIL("Label mismatch");
+                }
+            }
+        }
+    }
+
+    void check_label_match(const array<Float>& match_map, const table& left, const table& right) {
+        SECTION("label shape is expected") {
+            REQUIRE(left.get_row_count() == right.get_row_count());
+            REQUIRE(left.get_column_count() == right.get_column_count());
+            REQUIRE(left.get_column_count() == 1);
+        }
+        SECTION("label match is expected") {
+            const auto left_rows = row_accessor<const Float>(left).pull({ 0, -1 });
+            const auto right_rows = row_accessor<const Float>(right).pull({ 0, -1 });
+            for (std::int64_t i = 0; i < left_rows.get_count(); i++) {
+                const Float l = left_rows[i];
+                const Float r = right_rows[i];
+                if (l != match_map[r]) {
+                    CAPTURE(l, r, match_map[r]);
+                    FAIL("Label mismatch for mapped centroids");
+                }
+            }
+        }
+    }
+
+    void check_nans(const kmeans::train_result<>& result) {
+        const auto [centroids, labels, iteration_count] = unpack_result(result);
+
+        SECTION("there is no NaN in centroids") {
+            REQUIRE(te::has_no_nans(centroids));
+        }
+        SECTION("there is no NaN in labels") {
+            REQUIRE(te::has_no_nans(labels));
+        }
+    }
+
+    void check_nans(const kmeans::infer_result<>& result) {
+        const auto [labels, objective_function] = unpack_result(result);
+
+        SECTION("there is no NaN in objective function values") {
+            REQUIRE(!std::isnan(objective_function));
+        }
+        SECTION("there is no NaN in labels") {
+            REQUIRE(te::has_no_nans(labels));
+        }
+    }
+
+private:
+    static auto unpack_result(const kmeans::train_result<>& result) {
+        const auto centroids = result.get_model().get_centroids();
+        const auto labels = result.get_labels();
+        const auto iteration_count = result.get_iteration_count();
+        return std::make_tuple(centroids, labels, iteration_count);
+    }
+    static auto unpack_result(const kmeans::infer_result<>& result) {
+        const auto labels = result.get_labels();
+        const auto objective_function = result.get_objective_function_value();
+        return std::make_tuple(labels, objective_function);
+    }
+};
+
+using kmeans_types = COMBINE_TYPES((float, double), (kmeans::method::lloyd_dense));
+
+TEMPLATE_LIST_TEST_M(kmeans_batch_test,
+                     "kmeans degenerated test",
+                     "[kmeans][batch]",
+                     kmeans_types) {
+    using oneapi::dal::detail::empty_delete;
```

Comment:
It would be good to have short description of what is tested here. Test name is not very informative